code/modules/clothing/clothing.dm

```diff
@@ -198,6 +200,27 @@ BLIND     // can't see anything
 		var/mob/M = loc
 		M.update_inv_shoes()
 
+/obj/item/clothing/shoes/attackby(obj/item/I, mob/user, params)
+	if(!can_hold_knives)
+		return
+	if(held_knife)
+		user << "<span class='notice'>There's already something in [src].</span>"
+		return
+	if(istype(I, /obj/item/weapon/kitchen/knife) || istype(I, /obj/item/weapon/pen))//can hold both regular pens and energy daggers. made for your every-day tactical librarians/murderers.

```

Comment:
This istype just bothers me too much... Shoes that can hold stuff are rarer so it seems fair enough to allow something tiny to be stashed in there. Also keeps you from putting a butchers cleaver in your boot (ouch!).